tests/integration/semantics/src/test/java/org/apache/pulsar/tests/integration/functions/runtime/PulsarFunctionsRuntimeTest.java

```diff
@@ -33,22 +33,23 @@
 import org.apache.pulsar.tests.integration.functions.utils.CommandGenerator.Runtime;
 import org.apache.pulsar.tests.topologies.PulsarCluster;
 import org.testcontainers.containers.Container.ExecResult;
+import org.testng.annotations.Factory;
 import org.testng.annotations.Test;
 
 /**
  * The tests that run over different container mode.
  */
 public abstract class PulsarFunctionsRuntimeTest extends PulsarFunctionsTestBase {
 
-    protected enum ContainerFactory {
+    public enum FunctionProcessingMode {
```

Comment:
This is not *processing mode*. If you want to rename it, I would prefer renaming it to 'RuntimeFactory' to keep the naming consistent with what functions is using - https://github.com/apache/incubator-pulsar/blob/master/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeFactory.java